keps/sig-node/3857-rro-mounts/README.md

```diff
@@ -0,0 +1,1178 @@
+
+<!--
+What signals should users be paying attention to when the feature is young
+that might indicate a serious problem?
+-->
+
+Look for an event saying indicating RRO is not supported by the runtime.
+```console
+$ kubectl get events -o json -w
+...
+{
+    ...
+    "kind": "Event",
+    "message": "Error: RRONotSupported",
+    ...
+}
+...
+```
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+<!--
+Describe manual testing that was done and the outcomes.
+Longer term, we may want to require automated upgrade/rollback tests, but we
+are missing a bunch of machinery and tooling and can't do that now.
+-->
+(Will be revisited during beta)
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+<!--
+Even if applying deprecation policies, they may still surprise some users.
+-->
+No
+
+### Monitoring Requirements
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+###### How can an operator determine if the feature is in use by workloads?
+
+<!--
+Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
+checking if there are objects with field X set) may be a last resort. Avoid
+logs or events for this purpose.
+-->
+
+Yes if `recursiveReadOnly` is present in a Pod manifest.
+
+Can be inspected by running `kubectl get pods -A -o json | gron | grep recursiveReadOnly` .
+
+###### How can someone using this feature know that it is working for their instance?
+
+<!--
+For instance, if this is a pod-related feature, it should be possible to determine if the feature is functioning properly
+for each individual pod.
+Pick one more of these and delete the rest.
+Please describe all items visible to end users below with sufficient detail so that they can verify correct enablement
+and operation of this feature.
+Recall that end users cannot usually observe component logs or access metrics.
+-->
+
+- [X] API .status
+  - Condition name:  `volumeMountStatus.recursiveReadOnly`
+<!--
+- [ ] Other (treat as last resort)
+  - Details:
+-->
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+<!--
+This is your opportunity to define what "normal" quality of service looks like
+for a feature.
+
+It's impossible to provide comprehensive guidance, but at the very
+high level (needs more precise definitions) those may be things like:
+  - per-day percentage of API calls finishing with 5XX errors <= 1%
+  - 99% percentile over day of absolute value from (job creation time minus expected
+    job creation time) for cron job <= 10%
+  - 99.9% of /health requests per day finish with 200 code
+
+These goals will help you determine what you need to measure (SLIs) in the next
+question.
+-->
+
+It should work (or fail) deterministically.
```

Comment:
If someone is consuming the PRR section of this KEP, I think having some more specificity would be good here.  I'm not sure that this is clear enough as written.

Reply:
Reworded

> - `recursiveReadOnly=Enabled`:
>   100% of pods that were scheduled into a node must run with recursive read-only mounts,
>   or, 100% of them must fail to run.
> 
> - `recursiveReadOnly=IfPossible`:
>   100% of pods that were scheduled into a node must run with or without recursive read-only mounts
> 
> - `recursiveReadOnly=Disabled`, or unset:
>   100% of pods that were scheduled into a node must run without recursive read-only mounts